src/server/game/Movement/MotionMaster.cpp

```diff
@@ -479,11 +479,12 @@ void MotionMaster::MoveFall(uint32 id /*=0*/)
     if (std::fabs(_owner->GetPositionZ() - tz) < 0.1f)
         return;
 
+    _owner->AddUnitMovementFlag(MOVEMENTFLAG_FALLING);
```

Comment:
All changes made to this file are wrong. Creatures should never receive MOVEMENTFLAG_FALLING